### PR TITLE
Lab07 - Inter-Integrated Circuit

### PR DESCRIPTION
**Pull request by TeodoraMiu**
### Pull Request Overview

This pull request adds lab07 - I2C.

### Testing Strategy

This pull request was tested by running the examples locally.

### TODO or Help Wanted

This pull request still needs review.

### Build

- [x] Ran `npm build`.

### Author

Signed-off-by: Teodora Miu <teodora.miu@stud.fils.upb.ro>

### CODE REVIEW

File: docs/lab/07/index.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+3. Get the pressure and temperature readings from the sensor, **using I2C**. 
+- Write the `ctrl_meas` register with appropiate configuration values. You can find information on the contents you should write to this register at section 4.3.4 of the datasheet. (**2p**)
+- Read the raw pressure value and print it over the serial. (**2p**)
+- Read the raw temperature value and print it over the serial. (**1p**)
+
+:::tip
+Use what you learned in the last lab.
+:::
+
+4. Show the temperature and pressure values on the screen. (**2p**)
```
Comment (alexandruradovici) [inline docs/lab/07/index.md:263]: Can we add another exercise with USB? The purpose being to register a serial device? We can start from an embassy example that echoes the text and ask the students to change the name of the device and echo back the reversed text.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: docs/lab/07/index.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+:::info
+The `SDA` and `SCL` wires are never actually driven (set to `LOW`/`HIGH`) by the controller/peripherals. The line is controlled by either pulling the line low or releasing the line high. 
+
+When the line is *pulled down*, this means that it is tied to `GND` through a pull-down resistor. This electronically translates to `LOW`.
+
+When the line is *released*, or *pulled up*, this means that it ties back to `3V3` (which we can consider as being the "default" state of the wire) through a pull-up resistor. This electronically translates to `HIGH`.
+
+The pull-down resistor has a smaller resistance than the pull-up one, so every time a peripheral wants to write a 0 to the line, it changes the circuit of the bus so that it ties directly to ground.
+
+This is called *open-drain connection*. You can read more about how it works [here](https://www.ti.com/lit/an/sbaa565/sbaa565.pdf?ts=1712891793335#:~:text=I2C%20is%20a%20two%2Dwire,and%20receive%20commands%20and%20data.), at section 2.2.
```
Comment (DTeodor-Alexaandru) [inline docs/lab/07/index.md:43]: Open drain uses only pull up resistor and connects directly to the ground for logic LOW. Pull down resistors are used for push-pull configuration(ex: spi) and is recommanded only if there is another CS pin

**Review by DTeodor-Alexaandru — CHANGES_REQUESTED** [review]

**Review by DTeodor-Alexaandru — APPROVED** [review]

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: docs/lab/07/images/i2c_7bit_address_transmission.svg
[the remark this answers is not in the document]
Comment (TeodoraMiu) [reply docs/lab/07/images/i2c_7bit_address_transmission.svg:1]: I don't see any difference between the image I used and the one in the slides. There is an ack after every byte sent.

**Review by DTeodor-Alexaandru — APPROVED** [review]

File: docs/lab/07/images/i2c_7bit_address_transmission.svg
Comment (alexandruradovici) [inline docs/lab/07/images/i2c_7bit_address_transmission.svg:1]: There is an updated image in the slides, this one has an error, ack is read when CLK is high.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

**Review by alexandruradovici — APPROVED** [review]

**Review by DTeodor-Alexaandru — APPROVED** [review]

**Pull request merged**